nautobot/extras/models/models.py

```diff
@@ -944,6 +953,187 @@ def log(
+        default=0,
+        editable=False,
+        verbose_name="Total Run Count",
+        help_text="Running count of how many times the schedule has triggered the task",
+    )
+    date_changed = models.DateTimeField(
+        auto_now=True,
+        verbose_name="Last Modified",
+        help_text="Datetime that this PeriodicTask was last modified",
+    )
+    description = models.TextField(
+        blank=True,
+        verbose_name="Description",
+        help_text="Detailed description about the details of this Periodic Task",
+    )
+    user = models.ForeignKey(
+        to=settings.AUTH_USER_MODEL,
+        on_delete=models.SET_NULL,
+        related_name="+",
+        blank=True,
+        null=True,
+        help_text="User that requested the schedule",
+    )
+    approved_by_user = models.ForeignKey(
+        to=settings.AUTH_USER_MODEL,
+        on_delete=models.SET_NULL,
+        related_name="+",
+        blank=True,
+        null=True,
+        help_text="User that approved the schedule",
+    )
+    approval_required = models.BooleanField(default=False)
+    approved_at = models.DateTimeField(
+        auto_now=False,
+        auto_now_add=False,
+        editable=False,
+        blank=True,
+        null=True,
+        verbose_name="Approval date/time",
+        help_text="Datetime that the schedule was approved",
+    )
+
+    objects = ScheduledJobExtendedQuerySet.as_manager()
+    no_changes = False
+
+    def __str__(self):
+        return "{0.name}: {0.interval}".format(self)
+
+    def get_absolute_url(self):
+        return reverse("extras:scheduledjob", kwargs={"pk": self.pk})
+
+    def save(self, *args, **kwargs):
+        self.queue = self.queue or None
+        if not self.enabled:
+            self.last_run_at = None
+        elif not self.last_run_at:
+            # I'm not sure if this is a bug, or "works as designed", but if self.last_run_at is not set,
+            # the celery beat scheduler will never pick up a recurring job. One-off jobs work just fine though.
+            if self.interval in [
+                JobExecutionType.TYPE_HOURLY,
+                JobExecutionType.TYPE_DAILY,
+                JobExecutionType.TYPE_WEEKLY,
+            ]:
+                # A week is 7 days, otherwise the iteration is set to 1
+                multiplier = 7 if self.interval == JobExecutionType.TYPE_WEEKLY else 1
+                # Set the "last run at" time to one interval before the scheduled start time
+                self.last_run_at = self.start_time - timedelta(
+                    **{JobExecutionType.CELERY_INTERVAL_MAP[self.interval]: multiplier},
+                )
+
+        super().save(*args, **kwargs)
+
+    def clean(self):
+        """
+        Model Validation
+        """
+        if self.user and self.approved_by_user and self.user == self.approved_by_user:
+            raise ValidationError("The requesting and approving users cannot be the same")
```

Comment:
Possibly also error if `approved_by_user` is `None` but `approved_at` is not None, or vice versa?